src/main/scala/com/kinja/soy/Soy.scala

```diff
@@ -34,16 +31,20 @@ object Soy {
    * @param items List items which must have implicit SoyWrites available.
    * @return The soy list.
    */
-  def list(items: SoyValueWrapper*): SoyList =
-    SoyList(items.map(_.asInstanceOf[SoyValueWrapperImpl].field))
+  def list(items: SoyValueWrapper*): SoyList = SoyList(items.toSeq.map(unwrap))
 
   /**
    * Creates a SoyMap from key-value pairs.
    * @param items Key-value pairs. The values must have implicit SoyWrites available.
    * @return The soy map.
    */
-  def map(items: (String, SoyValueWrapper)*): SoyMap =
-    SoyMap(items.map { case (k, v) => (k, v.asInstanceOf[SoyValueWrapperImpl].field) }.toMap)
+  def map(items: (String, SoyValueWrapper)*): SoyMap = SoyMap(items.map { case (k, v) => (k, unwrap(v)) }.toMap)
+
+  // Passed nulls will typecheck without needing the implicit conversion, so they need to be checked at runtime
+  private def unwrap(wrapper: SoyValueWrapper) = wrapper match {
+    case null => SoyNull
+    case SoyValueWrapperImpl(value) => value
```

Comment:
Not sure why we have both `SoyValueWrapper` and `SoyValueWrapperImpl`.

Reply:
That's to support the automatic conversion of heterogenous items done when you call `Soy.list(...)` and `Soy.map(...)`. For the record, this is entirely based on the Play JSON library: https://github.com/playframework/play-json/blob/master/play-json/shared/src/main/scala/play/api/libs/json/Json.scala#L197-L234